examples/vault-cluster-authentication-iam/README.md

```diff
@@ -1,36 +1,174 @@
+access the cluster.
+
+
+## Running this example
+
+You will need to create a [Google Image][google_image] that has Vault and Consul
+installed, which you can do using the [vault-consul-image example][image_example].
+All the GCE Instances in this example (including the GCE Instance that authenticates
+to Vault) install [Dnsmasq][dnsmasq] (via the [install-dnsmasq module][dnsmasq_module])
+so that all DNS queries for `*.consul` will be directed to the Consul Server cluster.
+Because Consul has knowledge of all the Vault nodes (and in some cases, of other
+services as well), this setup allows the instances to use Consul's DNS server for
+service discovery, and thereby to discover the IP addresses of the Vault nodes.
 
-To deploy a Vault Cluster:
+
+## Quick start
 
 1. `git clone` this repo to your computer.
-1. Build a Vault and Consul Google Image. See the [vault-consul-image example](https://github.com/hashicorp/terraform-google-vault/tree/master/examples/vault-consul-image) documentation
-   for instructions. Make sure to note down the ID of the Google Image.
+1. Build a Vault and Consul Google Image. See the [vault-consul-image example][image_example]
+  documentation for instructions. Make sure to note down the ID of the Google Image.
 1. Install [Terraform](https://www.terraform.io/).
 1. Make sure you local environment is authenticated to Google Cloud.
-1. Open `variables.tf` and fill in any variables that don't have a default, including putting your Google Image ID into
-   the `vault_source_image` and `consul_server_source_image` variables.
+1. Open `variables.tf` and fill in any variables that don't have a default, including
+  putting your Google Image ID into the `vault_source_image` and `consul_server_source_image`
+  variables.
 1. Run `terraform init`.
 1. Run `terraform plan`.
 1. If the plan looks good, run `terraform apply`.
+1. Run `curl <web_client_public_ip>:8080` to check if the web server in the client
+instance is fetching the secret from Vault correctly.
+
+## IAM Auth
+
+IAM auth is a process in which Vault leverages Google Cloud to identify and
+authorize the IAM Service Account that originates the login request. Although
+this method can be used with a Service Account attached to a GCE instance, Vault
+also provides authentication based specifically on GCE Instance metadata. To
+see more about the GCE method, please refer to the
+[vault-cluster-authentication-gce example][gce_example].
+
+The workflow is that the client trying to authenticate itself will need to
+communicate with the Google API to generate a signed [JSON Web Token (JWT)][jwt],
+which is a JSON-based open standard for creating access tokens. This process can
+be quite cumbersome, but the Vault cli tool can do that for you. If you wish to
+[generate the JWT yourself][generate_jwt] you can also use `curl` and `oauth2l`
+or the `gcloud` tool. Once with signed JWT, the client can send it in its login
```

Comment:
s/Once with signed JWT, the client can send/Once the client has a signed JWT, it can send/